xarray/core/dataset.py

```diff
@@ -1604,6 +1604,12 @@ def to_zarr(
         References
         ----------
         https://zarr.readthedocs.io/
+
+        Notes
+        -----
+        zarr may automatically chunk a DataArray if it is not chunked or
+        if `chunks` is not set to -1 in its `enconding` attribute or in the
+        argument of the `encoding` parameter.
```

Comment:
Can we instead state what _does_ happen if a variable is already chunked or `chunks` can be found in `encoding`?

Something like

- If `chunks` are found in the `encoding` argument or attribute corresponding to any DataArray, those chunks are used.
- Otherwise, if the DataArray is already a dask array, it is written with those chunks.
- Finally, if not other chunks are found, Zarr uses its own heuristics to choose automatic chunk sizes.

Reply:
I closely followed your suggestion.
Note that I am not sure about the docstring style for a list-like note like this one.